### PR TITLE
MediaLibrary: MediaURLExporter and tests.

### PR DESCRIPTION
**Pull request by kurzee**
Continuing from #7118, this PR introduces the `MediaURLExporter` which primarily handles the detection of what the file at the URL actually is and exporting as needed.

Similarly to `MediaImageExporter`, errors are propagated via `throw` to keep track of exactly what went wrong along the way.

This exporter uses a value type for `URLExport` as the resulting export for the file depends on type of file it actually was.

- For images, we let the `MediaImageExporter` handle the export, if detected.
- For videos, we reuse the same kind code seen from the current export implementation in `develop` via `AVAssetExportSession`.
- For gifs, we just copy the file to our ideal Media directory.

Note: this exporter will be handy with the upcoming share extension.

To test:
1. Make sure the code compiles.
2. Run the tests for `MediaURLExporterTests.swift`

Needs review: @frosty and @SergioEstevao round four, final round 🤕😄

### CODE REVIEW

File: WordPress/Classes/Utility/Media/MediaURLExporter.swift [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    /// Enumerable type value for a URLExport, typed according to the resulting export of the file at the URL.
+    ///
+    public enum URLExport {
+        case exportedImage(MediaImageExport)
+        case exportedVideo(MediaVideoExport)
+        case exportedGIF(MediaGIFExport)
+    }
+
+    public enum ExportError: MediaExportError {
```
Comment (frosty) [inline WordPress/Classes/Utility/Media/MediaURLExporter.swift]: Should we name this `URLExportError`?

**Comment by kurzee** [comment]
>My only other question – should we handle non image / video / gif files here too? They'd just follow the same path as GIF – copy the file into place, I guess.

Yep! We certainly can, especially when we start bringing in "documents" from other sources, such as the share extension. For now though, I just wrote it to only support what we already do via current code.

**Comment by kurzee** [comment]
Ready for another look 👨‍🎤

File: WordPress/Classes/Utility/Media/MediaURLExporter.swift [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            onCompletion(URLExport.exportedGIF(MediaGIFExport(url: mediaURL,
+                                                              fileSize: fileSizeAtURL(mediaURL))))
+        } catch {
+            onError(exporterErrorWith(error: error))
+        }
+    }
+
+    /// Resolves the uniform type identifier for the file at the URL, or throws an error if unknown.
+    ///
+    fileprivate func typeIdentifierAtURL(_ url: URL) throws -> String {
```
Comment (SergioEstevao) [inline WordPress/Classes/Utility/Media/MediaURLExporter.swift]: Couldn't this be be an extension of the URL class?

**Pull request merged**